pkg/scheduler/scheduler.go

```diff
@@ -335,10 +378,7 @@ func (k *KubernetesScheduler) ResourceOffers(driver bindings.SchedulerDriver, of
 	k.offers.Add(offers)
 	for _, offer := range offers {
 		slaveId := offer.GetSlaveId().GetValue()
-		_, exists := k.slaves[slaveId]
-		if !exists {
-			k.slaves[slaveId] = newSlave(offer.GetHostname())
-		}
+		k.slaves.checkAndAdd(slaveId, offer.GetHostname())
```

Comment:
you can safely remove the lock/unlock that appears a few lines back -- it's no longer needed